packages/vscode-extension/src/webview/components/RenderOutlinesOverlay.tsx

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useRef } from "react";
+import { CanvasOutlineRenderer, OutlineRenderer, WorkerOutlineRenderer } from "react-scan";
+import {
+  RenderOutlinesEventListener,
+  RenderOutlinesEventMap,
+  RenderOutlinesInterface,
+} from "../../common/RenderOutlines";
+import { makeProxy } from "../utilities/rpc";
+import "./RenderOutlinesOverlay.css";
+
+const RenderOutlines = makeProxy<RenderOutlinesInterface>("RenderOutlines");
+
+function getDpr() {
+  return Math.min(window.devicePixelRatio || 1, 2);
+}
+
+interface Size {
+  width: number;
+  height: number;
+}
+
+function createOutlineRenderer(canvas: HTMLCanvasElement, size: Size, dpr: number) {
+  try {
+    return new WorkerOutlineRenderer(canvas, size, dpr);
+  } catch {
+    return new CanvasOutlineRenderer(canvas, size, dpr);
+  }
+}
+
+function RenderOutlinesOverlay() {
+  const hostRef = useRef<HTMLDivElement>(null);
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+  const outlineRendererRef = useRef<OutlineRenderer | null>(null);
+
+  useEffect(() => {
+    const host = hostRef.current;
+    const canvasEl = canvasRef.current;
+    if (!host || !canvasEl) {
+      return;
+    }
+
+    const dpr = getDpr();
+    let size = { width: hostRef.current.clientWidth, height: host.clientHeight };
```

Comment:
use `host` in both places here or `hostRef` in both places?

Reply:
How did that even happen 🤔